diff --git a/src/internet_identity/internet_identity.did b/src/internet_identity/internet_identity.did
index 0164427a..f8aa40a1 100644
--- a/src/internet_identity/internet_identity.did
+++ b/src/internet_identity/internet_identity.did
@@ -542,6 +542,11 @@ service : (opt InternetIdentityInit) -> {
     fetch_entries: () -> (vec BufferedArchiveEntry);
     acknowledge_entries: (sequence_number: nat64) -> ();
 
+    // Calls used for event stats housekeeping.
+    // Only callable by the canister itself.
+    prune_events_if_necessary: () -> ();
+    inject_prune_event: (timestamp: Timestamp) -> ();
+
     // V2 API
     // WARNING: The following methods are experimental and may change in the future.
 
