diff --git a/src/internet_identity/internet_identity.did b/src/internet_identity/internet_identity.did
index a47af160..4d75a7fe 100644
--- a/src/internet_identity/internet_identity.did
+++ b/src/internet_identity/internet_identity.did
@@ -171,6 +171,8 @@ type InternetIdentityStats = record {
     };
     archive_info: ArchiveInfo;
     canister_creation_cycles_cost: nat64;
+    max_num_latest_delegation_origins: nat64;
+    latest_delegation_origins: vec FrontendHostname;
     // Map from event aggregation to a sorted list of top 100 sub-keys to their weights.
     // Example: {"prepare_delegation_count 24h ic0.app": [{"https://dapp.com", 100}, {"https://dapp2.com", 50}]}
     event_aggregations: vec record {text; vec record {text; nat64}};
